### PR TITLE
WIP: Use MinGW build envs in Travis.

### PR DESCRIPTION
**Pull request by mkrautz**

### CODE REVIEW

File: scripts/travis-ci/before_install.bash [hunk 1]
```diff
@@ -5,14 +5,45 @@
...
 		sudo apt-get build-dep -qq mumble
-	elif [ "${MUMBLE_QT}" == "qt5" ]; then
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "x86_64-linux-gnu" ]; then
 		sudo apt-get -qq update
 		sudo apt-get build-dep -qq mumble
 		sudo apt-get install qt5-default qttools5-dev qttools5-dev-tools qtbase5-dev qtbase5-dev-tools qttranslations5-l10n libqt5svg5-dev
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "i686-w64-mingw32" ]; then
+		sudo dpkg --add-architecture i386
+		sudo apt-get -qq update
+		echo "deb http://pkg.mxe.cc/repos/apt/debian wheezy main" | sudo tee /etc/apt/sources.list.d/mxeapt.list
```
Comment (davidebeatrici) [inline scripts/travis-ci/before_install.bash]: We are using Debian Jessie, right?
If we are, we should use `deb http://pkg.mxe.cc/repos/apt/debian jessie main`.

File: scripts/travis-ci/before_install.bash [hunk 2]
```diff
@@ -5,14 +5,45 @@
...
+			mxe-${MUMBLE_HOST_DEB}.static-boost \
+			mxe-${MUMBLE_HOST_DEB}.static-protobuf \
+			mxe-${MUMBLE_HOST_DEB}.static-sqlite \
+			mxe-${MUMBLE_HOST_DEB}.static-flac \
+			mxe-${MUMBLE_HOST_DEB}.static-ogg \
+			mxe-${MUMBLE_HOST_DEB}.static-vorbis \
+			mxe-${MUMBLE_HOST_DEB}.static-libsndfile
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "x86_64-w64-mingw32" ]; then
+		sudo apt-get -qq update
+		echo "deb http://pkg.mxe.cc/repos/apt/debian wheezy main" | sudo tee /etc/apt/sources.list.d/mxeapt.list
```
Comment (davidebeatrici) [inline scripts/travis-ci/before_install.bash]: Same as above.

File: scripts/travis-ci/script.bash [hunk 3]
```diff
@@ -6,10 +6,18 @@
...
 
 if [ "${TRAVIS_OS_NAME}" == "linux" ]; then
-	if [ "${MUMBLE_QT}" == "qt4" ]; then
+	if [ "${MUMBLE_QT}" == "qt4" ] && [ "${MUMBLE_HOST}" == "x86_64-linux-gnu" ]; then
 		qmake-qt4 CONFIG+="release tests g15-emulator qt4-legacy-compat" -recursive && make -j2 && make check
-	elif [ "${MUMBLE_QT}" == "qt5" ]; then
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "x86_64-linux-gnu" ]; then
 		qmake CONFIG+="release tests g15-emulator" -recursive && make -j2 && make check
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "i686-w64-mingw32" ]; then
+		/usr/lib/mxe/usr/bin/mxe-${MUMBLE_HOST}.static-qmake -recursive -Wall CONFIG+="release tests g15-emulator"
```
Comment (davidebeatrici) [inline scripts/travis-ci/script.bash]: Should be `/usr/lib/mxe/usr/bin/${MUMBLE_HOST}.static-qmake -recursive -Wall CONFIG+="release tests g15-emulator"`

File: scripts/travis-ci/script.bash [hunk 4]
```diff
@@ -6,10 +6,18 @@
...
+		make -j2
+		make check
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "x86_64-w64-mingw32" ]; then
+		/usr/lib/mxe/usr/bin/mxe-${MUMBLE_HOST}.static-qmake -recursive -Wall CONFIG+="release tests g15-emulator"
```
Comment (davidebeatrici) [inline scripts/travis-ci/script.bash]: Same as above.

File: scripts/travis-ci/script.bash [hunk 5]
```diff
@@ -6,10 +6,18 @@
...
 
 if [ "${TRAVIS_OS_NAME}" == "linux" ]; then
-	if [ "${MUMBLE_QT}" == "qt4" ]; then
+	if [ "${MUMBLE_QT}" == "qt4" ] && [ "${MUMBLE_HOST}" == "x86_64-linux-gnu" ]; then
 		qmake-qt4 CONFIG+="release tests g15-emulator qt4-legacy-compat" -recursive && make -j2 && make check
-	elif [ "${MUMBLE_QT}" == "qt5" ]; then
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "x86_64-linux-gnu" ]; then
 		qmake CONFIG+="release tests g15-emulator" -recursive && make -j2 && make check
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "i686-w64-mingw32" ]; then
+		/usr/lib/mxe/usr/bin/${MUMBLE_HOST}.static-qmake -recursive -Wall CONFIG+="release tests g15-emulator"
```
Comment (davidebeatrici) [inline scripts/travis-ci/script.bash]: Should be:
`/usr/lib/mxe/usr/bin/${MUMBLE_HOST}.static-qmake-qt5 -recursive -Wall CONFIG+="release tests g15-emulator"`
I didn't notice that "qt5" was missing during the previous review.

File: scripts/travis-ci/script.bash [hunk 6]
```diff
@@ -6,10 +6,18 @@
...
+		make -j2
+		make check
+	elif [ "${MUMBLE_QT}" == "qt5" ] && [ "${MUMBLE_HOST}" == "x86_64-w64-mingw32" ]; then
+		/usr/lib/mxe/usr/bin/${MUMBLE_HOST}.static-qmake -recursive -Wall CONFIG+="release tests g15-emulator"
```
Comment (davidebeatrici) [inline scripts/travis-ci/script.bash]: Same as above.

File: scripts/travis-ci/before_install.bash (hunk 1 above)
Comment (mkrautz) [inline scripts/travis-ci/before_install.bash]: We're actually on Ubuntu trusty.  Everything seems to work, though!

File: scripts/travis-ci/before_install.bash (hunk 2 above)
Comment (mkrautz) [inline scripts/travis-ci/before_install.bash]: We're actually on Ubuntu trusty.  Everything seems to work, though!

File: scripts/travis-ci/before_install.bash (hunk 1 above)
Comment (davidebeatrici) [inline scripts/travis-ci/before_install.bash]: Oh, OK!
The packages should be equal for both Jessie and Wheezy, but I would use the Jessie ones anyway.

File: scripts/travis-ci/before_install.bash (hunk 2 above)
Comment (davidebeatrici) [inline scripts/travis-ci/before_install.bash]: Oh, OK!
The packages should be equal for both Jessie and Wheezy, but I would use the Jessie ones anyway.

File: scripts/travis-ci/before_install.bash (hunk 1 above)
Comment (mkrautz) [inline scripts/travis-ci/before_install.bash]: Well, I can try. But I think trusty is older than jessie, so the binaries might require a newer libc.

File: scripts/travis-ci/before_install.bash (hunk 2 above)
Comment (mkrautz) [inline scripts/travis-ci/before_install.bash]: Well, I can try. But I think trusty is older than jessie, so the binaries might require a newer libc.

**Comment by mkrautz** [comment]
Superceeded by @davidebeatrici's MinGW PR.

**Pull request closed without merging**